src/actions/view_actions.ts

```diff
@@ -217,6 +218,64 @@ export const createRemoveFilter: ActionSpec = {
   icon: "o-spreadsheet-Icon.FILTER_ICON_INACTIVE",
 };
 
+export const groupColumns: ActionSpec = {
+  name: (env) => {
+    const selection = env.model.getters.getSelectedZone();
+    if (selection.left === selection.right) {
+      return _lt("Group column %s", numberToLetters(selection.left));
+    }
+    return _lt(
+      "Group columns %s - %s",
+      numberToLetters(selection.left),
+      numberToLetters(selection.right)
+    );
+  },
+  execute: (env) => groupHeadersAction(env, "COL"),
+  isVisible: isSingleZoneSelected,
+  icon: "o-spreadsheet-Icon.GROUP_COLUMNS",
+};
+
+export const groupRows: ActionSpec = {
+  name: (env) => {
+    const selection = env.model.getters.getSelectedZone();
+    if (selection.top === selection.bottom) {
+      return _lt("Group row %s", String(selection.top + 1));
+    }
+    return _lt("Group rows %s - %s", String(selection.top + 1), String(selection.bottom + 1));
+  },
+  execute: (env) => groupHeadersAction(env, "ROW"),
+  isVisible: isSingleZoneSelected,
```

Comment:
Having both group & ungroup is relevant because of all the magic. I can disable the "group" item when there is already a group with this selection tho.